spec/controllers/post_controller_spec.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe PostsController, type: :controller do
+  describe '#index' do
+    before do
+      @user = FactoryBot.create(:user)
+    end
+
+    context 'as an authenticated user' do
+      it 'responds successfully' do
+        sign_in @user
```

Comment:
If you have multiple cases where `sign_in @user` is necessary within a context etc. you can do 
```ruby
before do
  sign_in @user
end
```
within the context of context or describe block

[OPTIONAL ADVICE]

Reply:
Yeah, there are so many places in the test we could prevent `repeating the code`. This will be refactored in the coming milestone.